ansible_collections/arista/avd/roles/eos_designs/python_modules/inband_management/avdstructuredconfig.py

```diff
@@ -0,0 +1,221 @@
+        return get(self._hostvars, "switch.mlag") is True
+
+    @cached_property
+    def _mlag_role(self) -> str | None:
+        return get(self._hostvars, "switch.mlag_role")
+
+    @cached_property
+    def vlans(self) -> dict | None:
+        VLAN_CFG = {
+            "tenant": "system",
+            "name": "L2LEAF_INBAND_MGMT",
+        }
+
+        if self._inband_management_role == "child":
+            return {self._inband_management_vlan: VLAN_CFG}
+
+        if self._inband_management_role == "parent":
+            return {vlan: VLAN_CFG for vlan in self._inband_management_data["vlans"]}
+
+        return None
+
+    @cached_property
+    def management_interfaces(self) -> dict | None:
+        if self._inband_management_role != "child":
+            return None
+
+        return {
+            self._inband_management_interface: {
+                "description": "L2LEAF_INBAND_MGMT",
+                "shutdown": False,
+                "mtu": self._p2p_uplinks_mtu,
+                "ip_address": self._inband_management_ip,
+                "gateway": self._inband_management_gateway,
+                "type": "inband",
+            }
+        }
+
+    @cached_property
+    def static_routes(self) -> dict | None:
+        if self._inband_management_role != "child":
+            return None
+
+        return [
+            {
+                "destination_address_prefix": "0.0.0.0/0",
+                "gateway": self._inband_management_gateway,
+            }
+        ]
+
+    @cached_property
+    def _inband_management_data(self) -> dict:
+        vlans = []
+        subnets = []
+        peers = get(self._hostvars, f"avd_topology_peers..{self._hostname}", separator="..", default=[])
+        avd_switch_facts = get(self._hostvars, "avd_switch_facts", required=True)
+        for peer in peers:
+            if self._hostname not in get(avd_switch_facts, f"{peer}..switch..inband_management_parents", separator="..", default=[]):
+                continue
+
+            if (subnet := get(avd_switch_facts, f"{peer}..switch..inband_management_subnet", separator="..", required=True)) in subnets:
+                continue
+
+            subnets.append(subnet)
+            vlans.append(get(avd_switch_facts, f"{peer}..switch..inband_management_vlan", separator="..", required=True))
+
+        if not subnets:
+            return {}
+
+        return {
+            "vlans": vlans,
+            "subnets": subnets,
+        }
+
+    @cached_property
+    def vlan_interfaces(self) -> dict | None:
+        if self._inband_management_role != "parent":
+            return None
+
+        vlan_interfaces = {}
+        for index, subnet in enumerate(self._inband_management_data["subnets"]):
+            vlan = self._inband_management_data["vlans"][index]
+            vlan_interface_name = f"Vlan{vlan}"
+            vlan_interfaces[vlan_interface_name] = self._get_svi_cfg(subnet)
+
+        if vlan_interfaces:
+            return vlan_interfaces
```

Comment:
Based on above suggestion.
```suggestion
        return vlan_interfaces
```